### PR TITLE
chore: migrate to owl bot

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-error-reporting/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Comment by SurferJeffAtGoogle** [comment]
@bcoe `gts fix` failed.  Why?
```
Step #5: DEBUG:synthtool:Running fix...
Step #5: 
Step #5: > @google-cloud/error-reporting@2.0.1 fix /workspace/nodejs-error-reporting
Step #5: > gts fix
Step #5: 
Step #5: version: 14
Step #5: 
Step #5: /workspace/nodejs-error-reporting/samples/explicitSetup.js
Step #5:   26:36  error  "@google-cloud/error-reporting" is not found  node/no-missing-require
Step #5: 
Step #5: /workspace/nodejs-error-reporting/samples/express.js
Step #5:   28:36  error  "@google-cloud/error-reporting" is not found  node/no-missing-require
Step #5: 
Step #5: /workspace/nodejs-error-reporting/samples/implicitSetup.js
Step #5:   26:36  error  "@google-cloud/error-reporting" is not found  node/no-missing-require
Step #5: 
Step #5: /workspace/nodejs-error-reporting/samples/manual.js
Step #5:   26:36  error  "@google-cloud/error-reporting" is not found  node/no-missing-require
Step #5: 
Step #5: /workspace/nodejs-error-reporting/samples/quickstart.js
Step #5:   20:36  error  "@google-cloud/error-reporting" is not found  node/no-missing-require
Step #5: 
Step #5: ✖ 5 problems (5 errors, 0 warnings)
Step #5: 
Step #5: Error: Command failed with exit code 1: node ./node_modules/eslint/bin/eslint --fix **/*.ts **/*.js **/*.tsx **/*.jsx --no-error-on-unmatched-pattern
Step #5:     at makeError (/workspace/nodejs-error-reporting/node_modules/gts/node_modules/execa/lib/error.js:59:11)
Step #5:     at handlePromise (/workspace/nodejs-error-reporting/node_modules/gts/node_modules/execa/index.js:114:26)
Step #5:     at processTicksAndRejections (internal/process/task_queues.js:93:5)
Step #5:     at async run (/workspace/nodejs-error-reporting/node_modules/gts/build/src/cli.js:120:17) 
```

**Review by JustinBeckwith — APPROVED** [review]

**Comment by bcoe** [comment]
@SurferJeffAtGoogle it's expected behavior `fix` will fail on these checks, because we have not linked the samples directory (this specific check ensures that the top level library has been linked against the samples directory, and that all its dependencies are installed).

To avoid this breaking OwlBot, we run `fix_hermetic` with [`check` false](https://github.com/googleapis/synthtool/blob/master/synthtool/languages/node.py#L200). This makes me think that there must be something else breaking nodejs-error-reporting -- and the fix is perhaps noise in the logs.

**Comment by SurferJeffAtGoogle** [comment]
Acknowledged.

File: system-test/error-reporting.ts [hunk 1]
```diff
@@ -463,7 +463,7 @@ describe('error-reporting', () => {
 
     const initConfiguration = Object.assign(
       {
-        reportMode: 'always' as 'always',
+        reportMode: 'always' as const,
```
Comment (SurferJeffAtGoogle) [inline system-test/error-reporting.ts:466]: All tests fail now.  Is it something to do with this change?

**Review by bcoe — APPROVED** [review]

**Pull request merged**